src/uct/ib/rc/base/rc_iface.h

```diff
@@ -198,13 +198,13 @@ typedef void (*uct_rc_iface_cleanup_rx_func_t)(uct_rc_iface_t *iface);
 typedef ucs_status_t (*uct_rc_iface_fc_ctrl_func_t)(uct_ep_t *ep, unsigned op,
                                                     uct_rc_pending_req_t *req);
 
-typedef ucs_status_t (*uct_rc_iface_fc_handler_func_t)(uct_rc_iface_t *iface,
-                                                       unsigned qp_num,
-                                                       uct_rc_hdr_t *hdr,
-                                                       unsigned length,
-                                                       uint32_t imm_data,
-                                                       uint16_t lid,
-                                                       unsigned flags);
+/* Returns 1 if AM data packed along with FC information, otherwise - 0 */
+typedef int (*uct_rc_iface_fc_handler_func_t)(uct_rc_iface_t *iface,
```

Comment:
IMO better keep return status as ucs_status_t to align with AM handlers

Reply:
but then it will be always `UCS_OK`, since I moved AM handling from the function

Reply:
if it always returns OK, why can't it be void?

Reply:
> if it always returns OK, why can't it be void?

we return whether we should handle AM or not.
but we could decide it after handling FC - see fc63c25